src/components/modal/createModalLauncher.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Modal, ModalProps } from '@patternfly/react-core';
+
+export type ModalComponentProps = Omit<ModalProps, 'children' | 'isOpen' | 'appendTo'> & {
+  ref?: React.LegacyRef<Modal>;
+};
+
+export type ModalLauncher = (onClose: () => void) => React.ReactElement;
+
+export type CreateModalLauncher = <P extends ModalComponentProps>(
+  Component: React.ComponentType<P>,
+  modalConfig?: ModalComponentProps & {
+    'data-testid': string;
+  },
+) => (componentProps?: P) => ModalLauncher;
+
+export const createModalLauncher: CreateModalLauncher =
+  (Component, modalProps) => (props) => (onClose) => {
```

Comment:
Either keep it `modalConfig` or `modalProps` in both places. 
```suggestion
  (Component, modalProps) => (componentProps) => (onClose) => {
```